TwinePlatform/twine-monolith
# 👍 address projects/activities comments

fixes #404 

### Changes
- Add a couple more tests to improve coverage
- Persist legend state when switching back and forth between projects and activities
- Use different colours for charts on projects and activities page
- Get rid of `ProjectsTabs` 
  - This is to avoid passing props through an additional layer unnecessarily
  - The ternary switching the spinner in and out based on the value of `loading` can be replaced with a `LoadingBoundary` component
- `aggregatedToStackedGraphData` takes an optional `GraphColoursEnum[]` argument
- update colours

### Testing Requirements
  - Projects/Activities page
    - Switch back and forth between projects and activities, while making changes to the legend selection
    - Ensure any changes to a legend are still present when switching back to it
    - Note the colours of the bars in "projects" and "activities" views to ensure they are sufficiently different to avoid confusion

### Release Requirements
- PO Approval of the feature branch

### Manual Deployment
- Dashboard
